stock_quant_product_publish/wizards/stock_quant_product_publish_wizard.py

```diff
@@ -19,5 +19,6 @@ def action_stock_quant_publish(self):
             "website_published": True,
             "yahoo_product_state_id": self.yahoo_product_state_id.id,
         }
-        for quant in quants:
-            quant.product_id.product_tmpl_id.sudo().write(values)
+        # Using mapped to gather all product_tmpl_id records
+        product_templates = quants.mapped('product_id.product_tmpl_id')
+        product_templates.sudo().write(values)
```

Comment:
Shouldn't this suffice?
```suggestion
        quants.mapped("product_id").sudo().write(values)
```

Reply:
Actually, I'm a bit confused about the scope as stock_quant_status_update also seems to update `yahoo_product_state_id` of the product. Please check to make sure we are not duplicating functions.

Reply:
I checked with the customer and they can update `yahoo_product_state_id` alone. And they have an operation to update `yahoo_product_state_id` whenever they publish the website.
If each function were to stand alone, it need two operations to publish the product on the website, and it would be difficult to know which product's `yahoo_product_state_id` was updated.